lhotse/recipes/spgispeech.py

```diff
@@ -0,0 +1,166 @@
+"""
+Description taken from official website: https://datasets.kensho.com/datasets/spgispeech
+SPGISpeech consists of 5,000 hours of recorded company earnings calls and their respective 
+transcriptions. The original calls were split into slices ranging from 5 to 15 seconds in 
+length to allow easy training for speech recognition systems. Calls represent a broad 
+cross-section of international business English; SPGISpeech contains approximately 50,000 
+speakers, one of the largest numbers of any speech corpus, and offers a variety of L1 and 
+L2 English accents. The format of each WAV file is single channel, 16kHz, 16 bit audio.
+
+Transcription text represents the output of several stages of manual post-processing. 
+As such, the text contains polished English orthography following a detailed style guide, 
+including proper casing, punctuation, and denormalized non-standard words such as numbers 
+and acronyms, making SPGISpeech suited for training fully formatted end-to-end models.
+
+Official reference:
+
+O’Neill, P.K., Lavrukhin, V., Majumdar, S., Noroozi, V., Zhang, Y., Kuchaiev, O., Balam, 
+J., Dovzhenko, Y., Freyberg, K., Shulman, M.D., Ginsburg, B., Watanabe, S., & Kucsko, G. 
+(2021). SPGISpeech: 5, 000 hours of transcribed financial audio for fully formatted 
+end-to-end speech recognition. ArXiv, abs/2104.02014.
+
+ArXiv link: https://arxiv.org/abs/2104.02014
+"""
+import logging
+import string
+from pathlib import Path
+from typing import Dict, Union
+
+from tqdm.auto import tqdm
+
+from lhotse.audio import Recording, RecordingSet
+from lhotse.parallel import parallel_map
+from lhotse.recipes.utils import manifests_exist, read_manifests_if_cached
+from lhotse.supervision import SupervisionSet, SupervisionSegment
+from lhotse.utils import Pathlike, Seconds
+
+
+def download_spgispeech(
+    target_dir: Pathlike = ".",
+) -> None:
+    """
+    Download and untar the dataset.
+
+    NOTE: This function just returns with a message since SPGISpeech is not available
+    for direct download.
+
+    :param target_dir: Pathlike, the path of the dir to storage the dataset.
+    """
+    logging.info(
+        "SPGISpeech is not available for direct download. Please fill out the form at"
+        " https://datasets.kensho.com/datasets/spgispeech to download the corpus."
+    )
+
+
+def normalize(text: str) -> str:
+    # Remove all punctuation
+    text = text.translate(str.maketrans("", "", string.punctuation))
```

Comment:
are you sure that you also want to remove apostrophes and dashes?

Reply:
I followed the ESPNet recipe for [text normalization](https://github.com/espnet/espnet/blob/9c24b3adddbde3402530080cb58ae08a6f4dd642/egs2/spgispeech/asr1/local/data_prep.sh#L30), where they do:
```
awk '{for(x=2;x<=NF;x++){gsub(/[[:punct:]]/,"",$x)}}1'
```
I believe this removes everything including apostrophes and dashes.

Reply:
weird but fair enough

Reply:
can you at least add a comment about it?

Reply:
sure